discovery/discovery.go

```diff
@@ -82,7 +83,10 @@ func (o *orchestratorPool) GetOrchestrators(numOrchestrators int) ([]*net.Orches
 			orchInfos = append(orchInfos, info)
 			numSuccessResp++
 		}
-		if numSuccessResp >= numOrchestrators || numResp >= len(o.uris) {
```

Comment:
Long term, removing the `numSuccessResp >= numOrchestrators` check is problematic because not all orchestrators on the network will respond in a timely manner, which means we have to wait for a time-out before returning from O selection. This impacts stream start-up time and failover behavior.

Reply:
That's a good point. I'll add randomization to the initial list. Seems like an easier solution for now